smac/epm/base_epm.py

```diff
@@ -12,64 +15,188 @@
 class AbstractEPM(object):
     '''Abstract implementation of the EPM API '''
 
-    def __init__(self, rng):
+    def __init__(self, 
+                 instance_features:np.ndarray=None,
+                 pca_dims:float=None):
         '''
         initialize random number generator
 
         Parameters
         ----------
-        rng : np.random.RandomState
+        instance_features: np.ndarray (I, K)
+            Contains the K dimensional instance features
+            of the I different instances
+        pca_dims: float
+            if set to a float, use PCA to reduce dimensionality of instance features
+            also requires to set n_feats (> pca_dims)
```

Comment:
I think the parameter name of this parameter should more closely follow the parameter name of the scikit-learn parameter, as it is only a wrapper to the scikit-learn parameter `n_components`. Also, you can use `typing.Union[int, float]` to show that the parameter can be both int and float.